src/pages/resultsView/ResultsViewPageStoreUtils.spec.ts

```diff
@@ -0,0 +1,425 @@
+import {getSimplifiedMutationType} from "shared/lib/oql/accessors";
+import {assert} from "chai";
+import {Mutation} from "../../shared/api/generated/CBioPortalAPI";
+import {
+    annotateMutationPutativeDriver,
+    computeCustomDriverAnnotationReport, computePutativeDriverAnnotatedMutations,
+    initializeCustomDriverAnnotationSettings
+} from "./ResultsViewPageStoreUtils";
+import {observable} from "mobx";
+
+describe("ResultsViewPageStoreUtils", ()=>{
+    describe("computeCustomDriverAnnotationReport", ()=>{
+        let driverFilterMutation:Mutation;
+        let driverTiersFilterMutation:Mutation;
+        let bothMutation:Mutation;
+        let neitherMutation:Mutation;
+
+        before(()=>{
+            driverFilterMutation = {
+                driverFilter:"B"
+            } as Mutation;
+
+            driverTiersFilterMutation = {
+                driverTiersFilter:"T"
+            } as Mutation;
+
+            bothMutation = {
+                driverFilter:"ADFADF",
+                driverTiersFilter:"SDPOIFJP"
+            } as Mutation;
+
+            neitherMutation = {
+            } as Mutation;
+        });
+
+        it("returns the right report for empty list", ()=>{
```

Comment:
for discussion, but  i would argue only two tests are necessary for this function.  1. mutation list with element sans driverFilter and 2. mutation list where they all have driver filter.   it seems to me that would:
1. cover all logical branches. 
2. cover the format of the return value

I don't think we need to test empty array, because that's just testing that the for loop will properly handle a zero length array.  I.e. testing the javascript engine.  
@ersinciftci please comment

Reply:
I agree with @ersinciftci , thats how I was taught at school too - treat it as black box and cover "representative cases" to try to cover all functionality, including edge cases like empty and singleton.